### PR TITLE
[FLINK-35182] Bump org.apache.commons:commons-compress from 1.24.0 to 1.26.1 for Flink Pulsar connector

### PR DESCRIPTION
**Pull request by GOODBOY008**
This closes https://github.com/apache/flink-connector-pulsar/pull/83.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -397,6 +397,20 @@ under the License.
...
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-lang3</artifactId>
+                <version>3.14.0</version>
+            </dependency>
+
+            <!-- For dependency convergence  -->
+            <dependency>
+                <groupId>commons-io</groupId>
+                <artifactId>commons-io</artifactId>
+                <version>2.15.1</version>
```
Comment (syhily) [inline pom.xml:411]: Move the version to the properties.

File: pom.xml [hunk 2]
```diff
@@ -397,6 +397,20 @@ under the License.
                 <version>${commons-compress.version}</version>
             </dependency>
 
+            <!-- For dependency convergence  -->
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-lang3</artifactId>
+                <version>3.14.0</version>
```
Comment (syhily) [inline pom.xml:404]: Move the version to the properties.

**Review by syhily — CHANGES_REQUESTED** [review]

File: flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/SampleData.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public Foo() {}
 
         @Override
         public String toString() {
-            return "" + i + "," + f + "," + (bar == null ? "null" : bar.toString());
+            return i + "," + f + "," + (bar == null ? "null" : bar.toString());
```
Comment (syhily) [inline flink-connector-pulsar/src/test/java/org/apache/flink/connector/pulsar/testutils/SampleData.java:52]: I prefer to do only one thing in one PR. This refactor shouldn't be included in the version bump.

**Review by syhily — CHANGES_REQUESTED** [review]

**Comment by syhily** [comment]
OK, LGTM. We need to get the CI pass, finally.

**Review by syhily — APPROVED** [review]

**Comment by syhily** [comment]
@tisonkun

**Review by tisonkun — APPROVED** [review]
Thank you! This closes #83.

**Pull request merged**